sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/aio/_azureappconfigurationproviderasync.py

```diff
@@ -227,81 +220,63 @@ async def load(*args, **kwargs) -> "AzureAppConfigurationProvider":
-            except HttpResponseError as e:
-                if e.status_code == 404:
-                    # If the sentinel is not found a refresh should be triggered when it is created.
-                    logging.debug(
-                        "WatchKey key: %s label %s was configured but not found. Refresh will be triggered if created.",
-                        key,
-                        label,
-                    )
-                    provider._refresh_on[(key, label)] = None  # type: ignore
-                else:
-                    _delay_failure(start_time)
-                    raise e
-            except Exception as e:
-                _delay_failure(start_time)
-                raise e
     return provider
 
 
-def _buildprovider(
+async def _buildprovider(
     connection_string: Optional[str], endpoint: Optional[str], credential: Optional["AsyncTokenCredential"], **kwargs
 ) -> "AzureAppConfigurationProvider":
     # pylint:disable=protected-access
-    provider = AzureAppConfigurationProvider(**kwargs)
+    if connection_string:
+        endpoint = connection_string.split(";")[0].split("=")[1]
+    if not endpoint:
+        raise ValueError("No endpoint specified.")
     retry_total = kwargs.pop("retry_total", 2)
     retry_backoff_max = kwargs.pop("retry_backoff_max", 60)
+    replica_discovery_enabled = kwargs.pop("replica_discovery_enabled", True)
 
     if "user_agent" in kwargs:
         user_agent = kwargs.pop("user_agent") + " " + USER_AGENT
     else:
         user_agent = USER_AGENT
 
-    if connection_string:
-        provider._client = AzureAppConfigurationClient.from_connection_string(
-            connection_string,
-            user_agent=user_agent,
-            retry_total=retry_total,
-            retry_backoff_max=retry_backoff_max,
-            **kwargs
-        )
-        return provider
-    if endpoint is not None and credential is not None:
-        provider._client = AzureAppConfigurationClient(
-            endpoint,
-            credential,
-            user_agent=user_agent,
-            retry_total=retry_total,
-            retry_backoff_max=retry_backoff_max,
-            **kwargs
-        )
-        return provider
-    raise ValueError("Please pass either endpoint and credential, or a connection string.")
+    interval: int = kwargs.get("refresh_interval", 30)
+    if interval < 1:
+        raise ValueError("Refresh interval must be greater than or equal to 1 second.")
+
+    min_backoff: int = min(kwargs.pop("min_backoff", 30), interval)
+    max_backoff: int = min(kwargs.pop("max_backoff", 600), interval)
+
+    replica_client_manager = ConfigurationClientManager(
+        connection_string,
+        endpoint,
+        credential,
+        user_agent,
+        retry_total,
+        retry_backoff_max,
+        replica_discovery_enabled,
+        min_backoff,
+        max_backoff,
+        **kwargs
+    )
+    await replica_client_manager.setup_initial_clients()
```

Comment:
Why dont we need to setup failover endpoints in  the sync version?

Reply:
you can't have an `async __init__` in python. Setting up the clients is an async operation as we reach out to DNS.